src/Akeneo/CouplingDetector/Formatter/AbstractFormatter.php

```diff
@@ -0,0 +1,147 @@
+            Events::POST_NODES_PARSED  => 'postNodesParsed',
+            Events::PRE_RULES_CHECKED  => 'preRulesChecked',
+            Events::NODE_CHECKED       => 'nodeChecked',
+            Events::RULE_CHECKED       => 'ruleChecked',
+            Events::POST_RULES_CHECKED => 'postRulesChecked',
+        );
+    }
+
+    /**
+     * @param PreNodesParsedEvent $event
+     */
+    public function preNodesParsed(PreNodesParsedEvent $event)
+    {
+        $this->nodeCount = $event->getFinder()->count();
+        $this->outputPreNodesParsed($event);
+    }
+
+    /**
+     * @param NodeParsedEvent $event
+     */
+    public function nodeParsed(NodeParsedEvent $event)
+    {
+        $this->parsingNodeIteration++;
+        $this->outputNodeParsed($event);
+    }
+
+    /**
+     * @param PostNodesParsedEvent $event
+     */
+    public function postNodesParsed(PostNodesParsedEvent $event)
+    {
+        $this->outputPostNodesParsed($event);
+    }
+
+    /**
+     * @param PreRulesCheckedEvent $event
+     */
+    public function preRulesChecked(PreRulesCheckedEvent $event)
+    {
+        $this->ruleCount = count($event->getRules());
+        $this->outputPreRulesChecked($event);
+    }
+
+    /**
+     * @param NodeChecked $event
+     */
+    public function nodeChecked(NodeChecked $event)
+    {
+        $this->checkingNodeIteration++;
+        $key = $event->getNode()->getFilepath();
+        if (null !== $event->getViolation() && !in_array($key, $this->nodesOnError)) {
+            $this->nodesOnError[] = $key;
+        }
+
+        $this->outputNodeChecked($event);
+    }
+
+    /**
+     * @param RuleCheckedEvent $event
+     */
+    public function ruleChecked(RuleCheckedEvent $event)
+    {
+        $this->checkingRuleIteration++;
+        $this->checkingNodeIteration = 0;
+        $key = $event->getRule()->getSubject() . $event->getRule()->getType();
+        $nbErrors = count($event->getViolations());
+        if (0 !== $nbErrors && !in_array($key, $this->rulesOnError)) {
+            $this->rulesOnError[$key] = $event->getRule();
+            $this->violationsCount += $nbErrors;
+        }
+
+        $this->outputRuleChecked($event);
+    }
+
+    /**
+     * @param PostRulesCheckedEvent $event
+     */
+    public function postRulesChecked(PostRulesCheckedEvent $event)
+    {
+        $this->outputPostRulesChecked($event);
+    }
+
+    protected abstract function outputPreNodesParsed(PreNodesParsedEvent $event);
+    protected abstract function outputNodeParsed(NodeParsedEvent $event);
+    protected abstract function outputPostNodesParsed(PostNodesParsedEvent $event);
+    protected abstract function outputPreRulesChecked(PreRulesCheckedEvent $event);
+    protected abstract function outputNodeChecked(NodeChecked $event);
+    protected abstract function outputRuleChecked(RuleCheckedEvent $event);
+    protected abstract function outputPostRulesChecked(PostRulesCheckedEvent $event);
```

Comment:
Interesting way to define the contract that need to be implemented by your formatter.

I wonder if this should be the responsibility of another interface, let's say
```
interface FormatterInterface {
    public function outputPreNodesParsed(PreNodesParsedEvent $event);
    public function outputNodeParsed(NodeParsedEvent $event);
    public function outputPostNodesParsed(PostNodesParsedEvent $event);
    public function outputPreRulesChecked(PreRulesCheckedEvent $event);
    public function outputNodeChecked(NodeChecked $event);
    public function outputRuleChecked(RuleCheckedEvent $event);
    public function outputPostRulesChecked(PostRulesCheckedEvent $event);
}

An implementation of this interface should be injected in a FormatterSubscriber.

It looks a bit weird to me to say that a Formatter is a Subscriber in fact :)